app/src/main/kotlin/com/simplemobiletools/notes/activities/MainActivity.kt

```diff
@@ -48,6 +48,17 @@ class MainActivity : SimpleActivity(), ViewPager.OnPageChangeListener {
         pager_title_strip.layoutParams.height = (pager_title_strip.height + resources.getDimension(R.dimen.activity_margin) * 2).toInt()
         checkWhatsNewDialog()
         storeStoragePaths()
+
+        with(intent) {
+            if (action == Intent.ACTION_SEND && type != null) {
+                if (type.equals("text/plain")) {
+                    val text = getStringExtra(Intent.EXTRA_TEXT)
+                    text?.let {
+                        displayNewNoteDialog(it)
+                    }
+                }
+            }
+        }
```

Comment:
any particular difference between "with(intent)" and "intent.apply"? Im using apply everywhere. Also you can remove the local "text" variable and change the type check to 
` if (action == Intent.ACTION_SEND && type?.equals("text/plain") == true) {`